### DISCUSSION TITLE
Crop to path while preserving partial transparency

### DISCUSSION THREAD

**Discussion by zappzerapp** [ImageMagick/ImageMagick#6207] [Help] [answered]
### ImageMagick version

7.1.0-57

### Operating system, version and so on

macOS 13.2.1

### Description

Hello all,

I want to crop an image ([test.tif.zip](https://github.com/ImageMagick/ImageMagick/files/11109689/test.tif.zip)) to a path while preserving partial transparency.
My command looks like this:
`convert 'test.tif[0]' -alpha off -alpha transparent -clip-path 'path' -alpha opaque test.png`

On the left is the result and on the right would be the desired result:
![compare](https://user-images.githubusercontent.com/9962724/228782079-14a75cb1-9e41-406e-9f26-435fb1413178.png)

Thanks for the help!

**Comment by snibgo** [comment]
You use IM v7, so I suggest using `magick` so IM uses v7 syntax, not `convert` or `magick convert` which both use the old v6 syntax.

You ask to use a path "while preserving partial transparency". There is no partial transparency that can be preserved. I think you mean that you want anti-aliasing of the clip path.

Your command ...
```
convert 'test.tif[0]' -alpha off -alpha transparent -clip-path 'path' -alpha opaque test.png
```
... uses a binary clip path; it is either fully transparent or fully opaque.

To get an anti-aliased clipping path:
```
magick test.tif[0] -format "%[8BIM:1999,2998:#1]" info: | magick test.tif[0] ( svg:- -negate ) -alpha off -compose CopyOpacity -composite out.png
```

**Reply by zappzerapp** [reply]
Thanks for the hint with the syntax.
I want the "A" to be visible on the image, which unfortunately it is not with the anti-aliased variant either

![c](https://user-images.githubusercontent.com/9962724/228878281-c5b79bd9-caec-4a4b-81e5-42eef41fb651.png)

**Comment by snibgo** [comment, marked as answer]
I didn't see the "A" which is slightly transparent black. Instead of copying the opacity from the clip path to the image (which would overwrite the image's opacity), I multiply the two opacities, then copy that to the image.

The command gets a bit complex, so I split it into lines. Windows BAT syntax.
```
magick test.tif[0] -format "%%[8BIM:1999,2998:#1]" info: | magick ^
  test.tif[0] -write mpr:IMG +delete ^
  svg:- -write mpr:CLP +delete ^
  ( mpr:IMG ^
    -alpha extract ^
    -write mpr:ALPH +delete ^
  ) ^
  ( mpr:CLP -negate ^
    mpr:ALPH ^
    -alpha off ^
    -compose Multiply -composite ^
  ) ^
  mpr:IMG ^
  +swap ^
  -alpha off -compose CopyOpacity -composite ^
  out.png
```

**Reply by zappzerapp** [reply]
Looks like this will do exactly what I expect, thank you!

PS: The `%%` should be `%` , right?
